core/src/main/kotlin/org/evomaster/core/search/gene/ArrayGene.kt

```diff
@@ -103,87 +103,47 @@ class ArrayGene<T>(
         }
     }
 
-    override fun standardMutation(randomness: Randomness, apc: AdaptiveParameterControl, allGenes: List<Gene>) {
-
+    override fun candidatesInternalGenes(randomness: Randomness, apc: AdaptiveParameterControl, allGenes: List<Gene>, selectionStrategy: SubsetGeneSelectionStrategy, enableAdaptiveGeneMutation: Boolean, additionalGeneMutationInfo: AdditionalGeneSelectionInfo?): List<Gene> {
         if(!isMutable()){
             throw IllegalStateException("Cannot mutate a immutable array")
         }
+        if ( elements.isEmpty() || elements.size > maxSize){
+            return listOf()
+        }
+        val p = when(selectionStrategy){
+            SubsetGeneSelectionStrategy.ADAPTIVE_WEIGHT -> {
+                if(additionalGeneMutationInfo?.impact != null
+                        && additionalGeneMutationInfo.impact is ArrayGeneImpact
+                        && additionalGeneMutationInfo.impact.sizeImpact.noImprovement.any { it.value < 2 } //if there is recent improvement by manipulating size
+                ){
+                    0.3 // increase probability to mutate size
```

Comment:
this is a bit hard to to read. on one side, you have a constant 0.3, on the other `MODIFY_SIZE`.
0.3 should be its own variable (with meaningful name), or a derived from `MODIFY_SIZE`, eg
`x * MODIFY_SIZE` for some x value